modules/apps/foundation/configuration-admin/configuration-admin-web/src/main/java/com/liferay/configuration/admin/web/internal/portlet/action/BindConfigurationMVCActionCommand.java

```diff
@@ -128,7 +133,47 @@ public boolean processAction(
 			properties.put(ConfigurationAdmin.SERVICE_FACTORYPID, factoryPid);
 		}
 
-		configureTargetService(configurationModel, configuration, properties);
+		try {
+			configureTargetService(
+				configurationModel, configuration, properties);
+		}
+		catch (Exception e) {
+			Throwable cause = e.getCause();
+
+			while (cause != null) {
+				if (cause instanceof ConfigurationModelActionException) {
+					SessionErrors.add(actionRequest, cause.getClass());
+
+					PortletURL portletURL = PortletURLFactoryUtil.create(
+						actionRequest,
+						ConfigurationAdminPortletKeys.SYSTEM_SETTINGS,
+						themeDisplay.getPlid(), PortletRequest.RENDER_PHASE);
+
+					portletURL.setParameter(
+						"mvcRenderCommandName", "/edit_configuration");
+
+					String redirect = ParamUtil.getString(
+						actionRequest, "redirect");
+
+					portletURL.setParameter("redirect", redirect);
+
+					portletURL.setParameter("errorMessage", cause.getMessage());
```

Comment:
Why do we need to pass the error message here if it's put in SessionErrors? 

Also, putting it hear allows anyone to put any message in the UI with just some URL manipulation.